water_neuron/src/lib.rs

```diff
@@ -149,6 +153,89 @@ pub struct CanisterInfo {
     pub governance_fee_share_percent: u64,
 }
 
+#[derive(Debug, EnumString, EnumIter, Display)]
```

Comment:
I would put these types in another file `icrc21.rs`. I would also put the code of `icrc21_canister_call_consent_message` in this file. I don't like to have an implementation in `main.rs`.

Reply:
On top of that, it seems that the standard is unnecessarily heavy, but this is not related to your code changes.